### PR TITLE
Update Github Actions to have spelling and formatting checks

### PR DESCRIPTION
**Pull request by yourslab**
Update Github Actions to have spelling and formatting checks

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -60,3 +60,50 @@ jobs:
...
+            done
+  formatting:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Uncrustify
+        run: sudo apt-get install uncrustify
+      - name: Run Uncrustify
+        run: find . -iname "*.[hc]" -exec uncrustify --check -c tools/uncrustify.cfg {} +
+      - name: Check For Trailing Whitespace
```
Comment (aggarg) [inline .github/workflows/ci.yml:100]: Why do we need this separately?  Does uncrustify above not check it?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -60,3 +60,50 @@ jobs:
...
           if [[ "$(wc -c < doxyoutput.txt | bc)" = "0" ]]; then exit 0; else exit 1; fi
+  spell-check:
+      runs-on: ubuntu-latest
+      steps:
+        - name: Checkout Parent Repo
+          uses: actions/checkout@v2
+          with:
+            ref: master
+            repository: aws/aws-iot-device-sdk-embedded-C
+        - run: rm -r libraries/standard/coreJSON
```
Comment (aggarg) [inline .github/workflows/ci.yml:71]: `rm -rf`?

Comment (yourslab) [reply .github/workflows/ci.yml:71]: This script was copied from coreMQTT, but I don't think adding the -f will hurt. I have added it.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (yourslab) [reply .github/workflows/ci.yml:100]: Uncrustify will do it just fine for source files, but we still have cmake files to check. Interestingly, this caught a trailing whitespace in one of them.

**Review by aggarg — APPROVED** [review]

**Pull request merged**